### ISSUE TITLE
Growatt API Add Data

### ISSUE THREAD

**Issue by NathanealV** [Sjord/growatt_api_client#14]
How would I add more data to the JSON read out that is not already available there?

I see there are things like "PlantList.do" but where do I find these .do files or tags

Thank you so much for the API btw.

**Comment by Sjord** [comment]
I found the URLs by reverse engineering the Android app, and intercepting traffic between the Android app and the server. Is there particular data you are looking for? I recently added many [storage functions](https://github.com/Sjord/growatt_api_client/pull/13), these just need some more testing.

**Comment by NathanealV** [comment]
Would you be able to share the different URLs? Or share how you reverse engineered it? I guess I just need the entire stack of data that is there on the website - like the entire thing.

**Comment by NathanealV** [comment]
I have managed to find these URLs and have been getting the data that I need. Just one question, would you be able to help me with a json parsing error with one of the pages? It's with NewTwoPlantList.do and with parameter 
```
"op": "getUserCenterEnertyDataByPlantid",
```
Python seems to return an error parsing the json

**Comment by Sjord** [comment]
Did you provide the plantId and language parameters? Try this:

    def get_user_center_energy_data_by_plant_id(self, plant_id):
        response = self.session.post(
            self.get_url("newTwoPlantAPI.do"),
            params={"op": "getUserCenterEnertyDataByPlantid"},
            data={"plantId": plant_id, "language": 1},
        )
        return response.json()

**Comment by NathanealV** [comment]
Yes, that seems to work! Thank you for that.

**Comment by NathanealV** [comment]
Is there any way to login and complete a GET/POST request all in the same go/same url?

**Comment by Sjord** [comment]
No, I don't think so. You first need to login and then you can use the session to retrieve information.

**Comment by NathanealV** [comment]
Ah ok. I was planning to use a third-party interface that gets JSON data but it only provides 1 URL and so I don't know if I can do this. Thank you.
